compiler/rustc_middle/src/thir.rs

```diff
@@ -287,6 +287,9 @@ pub enum ExprKind<'tcx> {
         /// Note: in some cases (like calling a closure), the function call `f(...args)` gets
         /// rewritten as a call to a function trait method (e.g. `FnOnce::call_once(f, (...args))`).
         args: Box<[ExprId]>,
+        /// The spans of the args
+        /// (e.g. `a` and `b` in `x.foo(a, b)`).
+        arg_spans: Box<[Span]>,
```

Comment:
Can't you get the spans from the expr that the ExprId in the args above refers to?

And from that, you can eliminate all the thir and hir changes done in this PR?

Reply:
Indeed I can! When I tried to do it before, I couldn't figure out how to do it. But now when I tried again, it was easy, and makes the code much simpler. You were also right about !Copy being a left-over from adding arg_spans to HIR early on, and I forgot about it. Removed again now.